gateway/websocket_handlers.go

```diff
@@ -0,0 +1,106 @@
+package gateway
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	"github.com/buildwithgrove/path/protocol"
+	"github.com/buildwithgrove/path/websockets"
+)
+
+// ---------- Shannon Client Message Handler ----------
+
+var _ websockets.WebSocketMessageHandler = &websocketClientMessageHandler{}
+
+// websocketClientMessageHandler handles websocket client messages
+type websocketClientMessageHandler struct {
```

Comment:
This whole file should likely be removed, with different pieces moved to QoS, the WebSocket request context, etc.